src/components/MessagesList/MessagesGroup/Message/MessagePart/FilePreview.vue

```diff
@@ -534,6 +538,31 @@ export default {
 		},
 	},
 
+	watch: {
+		isTemporaryUpload: {
+			immediate: true,
+			handler(value) {
+				this.uploadManager = value ? getUploader() : null
+			}
+		},
+		'uploadFile.status'(value) {
+			if (['successUpload', 'sharing', 'shared'].includes(value)) {
+				this.uploadProgress = 100
+			}
+		},
+		upload: {
+			immediate: true,
+			deep: true,
+			handler(value) {
+				if (!value) {
+					return
+				}
+				const progress = value._uploaded / value._size * 100
+				this.uploadProgress = Math.max(this.uploadProgress, progress)
+			}
+		}
+	},
```

Comment:
This can be computed

```js
   uploadProgress() {
                                if (!this.upload) {
					return 100
				}
				return  this.upload._uploaded / this.upload._size * 100

		}
```

Unless ._uploaded is not progressing in one direction ( gets higher and then lower) or computed can't detect in depth object change here too despite we are using its attributes.

Reply:
It could fall back to 0 when upload got removed from the queue, but I will see what could be done

Reply:
Here is the lifecycle of upload:
```
(File) (fileUpload.status) (upload._progress/upload._size)
FilePreview.vue:379 initialised undefined
FilePreview.vue:379 pendingUpload undefined
FilePreview.vue:379 uploading 0
FilePreview.vue:379 uploading 0.025629074670996455
FilePreview.vue:379 uploading 0.05125814934199291
...
FilePreview.vue:379 uploading 0.9739048374978653
FilePreview.vue:379 uploading 0.9995339121688618
FilePreview.vue:379 uploading 1
FilePreview.vue:379 uploading undefined // progressBar shows 0 at this point, and it takes a while
FilePreview.vue:379 successUpload undefined
FilePreview.vue:379 sharing undefined
FilePreview.vue:379 shared undefined
```

Reply:
We can condition showing progressBar only if uploadProgress is not undefined. This means that this part is not included in the progress state :
`FilePreview.vue:379 uploading undefined // progressBar shows 0 at this point, and it takes a while`

Which makes sense because it is actually uploaded but fileUpload.status hasn't been updated yet ?